performance/performance.md

```diff
@@ -40,6 +40,15 @@ A good example is a social media widget. A plugin may call out to a third-party
 
 Persistent object caching can speed up WordPress and is offered my many managed WordPress hosts or can be set up independently. You will also need a plugin to connect WordPress to the object cache. There are many [available on the plugin directory](https://wordpress.org/plugins/search/object+cache/)
 
+### Op-code Caching (a.k.a server-side caching)
+As mentioned in the section on full page caching, WordPress processes PHP scripts as part of producing the HTML, CSS and JavaScript for a browser to load. It takes time for the web server to read each PHP script WordPress needs, to compile the script and to run the PHP script. By default, the web server has to do this process for every single visit to every single page. Full page caching can reduce how much the web server has to do this depending on what kind of full page caching is used, but there may still be some PHP scripts that have to be read, compiled, and run for every single visit.
+
+Having to read and compile PHP scripts for every single visit can be extremely taxing on a web server. It would be like if every time someone asked you a question, you had to read a book in a different language, translate that book into your own language, and then answer the question, without taking any notes or remembering what the book said after you answered the question.
+
+Op-code caching lets the web server take notes and remember what it has read between each time the web server is asked a question. Op-code caching stores a compiled copy of every PHP script in the server's memory (RAM). When the web server starts processing PHP scripts for WordPress, the web server checks the op-code cache for a cached copy of the PHP script. If there is a cached copy, the web server can skip straight to running the PHP script using the cached copy instead of having to read and compile the script again. Skipping reading and compiling PHP scripts can greatly improve the web server's resource usage and enable WordPress to serve many more requests than it might have been able to otherwise.
```

Comment:
Great definition! I think we should move this closer to the top, remove the analogy above, and add something like "Using op-code cache helps speed up your server since it can run WordPress without having to read and compile PHP scripts for every single page visit"